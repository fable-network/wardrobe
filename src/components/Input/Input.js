import React from 'react';
import styled from 'styled-components';
import { hexToRgba } from '../../helpers/colors';

const StyledInput = styled.input`
  box-sizing: border-box;
  width: 100%;
  padding: 8px 10px;
  border: solid 1px ${p => (p.disabled ? hexToRgba(p.theme.stoneGrey, 0.5) : p.theme.stoneGrey)};
  outline: none;
  color: ${p => (p.disabled ? hexToRgba(p.theme.stoneGrey, 0.5) : p.theme.ravenBlack)};
  font-size: 16px;
  text-overflow: ellipsis;
  cursor: ${p => (p.disabled ? 'not-allowed' : 'default')};

  ::placeholder {
    color: ${p => (p.disabled ? hexToRgba(p.theme.stoneGrey, 0.5) : p.theme.stoneGrey)};
  }

  &:focus {
    transition: 0.2s linear;
    border-color: ${p => p.theme.skyBlue};
    box-shadow: 0 0 3px ${p => p.theme.skyBlue};
  }

  ${p => p.invalid && `border-color: ${p.theme.flameRed}`};
`;

/**
 * Basic React `input` field with a bit of styling.
 */
const Input = props => <StyledInput {...props} />;

export default Input;
