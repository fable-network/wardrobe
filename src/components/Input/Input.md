Empty:

```jsx
<Input type="text" placeholder="Empty state" value="" />
```

Filled:

```jsx
<Input type="text" value="Some text" />
```

Disabled:

```jsx
<Input type="text" disabled placeholder="This input is disabled" />
```

Invalid:

```jsx
<Input
  type="text"
  placeholder="You could type something here"
  value="invalid value"
  invalid={true}
/>
```

Editable:

```jsx
<Input
  type="text"
  placeholder="You could type something here"
/>
```
