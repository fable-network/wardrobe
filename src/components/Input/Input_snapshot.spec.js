/* eslint-disable */
// This file was automatically generated
import React from 'react';
import 'jest-styled-components';
import Input from './Input';


jest.mock('../../charts/HighChart', () => () => <high-charts />);

describe('Input Component', () => {
  describe('snapshots', () => {
    it('should match snapshot', () => {
    
      const wrapper1 = mount(
        <Input type="text" placeholder="Empty state" value="" onChange={() => null} />
      );
      expect(wrapper1).toMatchSnapshot();

      const wrapper2 = mount(
        <Input type="text" value="Some text" onChange={() => null} />
      );
      expect(wrapper2).toMatchSnapshot();

      const wrapper3 = mount(
        <Input type="text" disabled placeholder="This input is disabled" />
      );
      expect(wrapper3).toMatchSnapshot();

      const wrapper4 = mount(
        <Input
          type="text"
          placeholder="You could type something here"
          value="invalid value"
          onChange={() => null}
          invalid={true}
        />
      );
      expect(wrapper4).toMatchSnapshot();

      const wrapper5 = mount(
        <Input
          type="text"
          placeholder="You could type something here"
        />
      );
      expect(wrapper5).toMatchSnapshot();

    });
  });
});
