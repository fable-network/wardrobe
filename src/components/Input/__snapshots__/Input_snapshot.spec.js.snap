// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component snapshots should match snapshot 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  padding-left: calc(0.5714285714285714rem - 1px);
  padding-right: calc(0.5714285714285714rem - 1px);
  padding-top: calc(0.625rem - 1px);
  padding-bottom: calc(0.625rem - 1px);
  border: solid 1px #d2d9dd;
  border-radius: 0.25rem;
  background-color: #ffffff;
  outline: none;
  color: #2d3b43;
  font-size: 0.875rem;
  line-height: 1.1428571428571428;
  text-overflow: ellipsis;
  cursor: text;
}

.c0::-webkit-input-placeholder {
  color: #949ea3;
}

.c0::-moz-placeholder {
  color: #949ea3;
}

.c0:-ms-input-placeholder {
  color: #949ea3;
}

.c0::placeholder {
  color: #949ea3;
}

.c0:focus {
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
  border-color: #146292;
  box-shadow: 0 0 2px #146292 inset;
}

<Input__StyledInput
  onChange={[Function]}
  placeholder="Empty state"
  type="text"
  value=""
>
  <input
    className="c0"
    onChange={[Function]}
    placeholder="Empty state"
    type="text"
    value=""
  />
</Input__StyledInput>
`;

exports[`Input Component snapshots should match snapshot 2`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  padding-left: calc(0.5714285714285714rem - 1px);
  padding-right: calc(0.5714285714285714rem - 1px);
  padding-top: calc(0.625rem - 1px);
  padding-bottom: calc(0.625rem - 1px);
  border: solid 1px #d2d9dd;
  border-radius: 0.25rem;
  background-color: #ffffff;
  outline: none;
  color: #2d3b43;
  font-size: 0.875rem;
  line-height: 1.1428571428571428;
  text-overflow: ellipsis;
  cursor: text;
}

.c0::-webkit-input-placeholder {
  color: #949ea3;
}

.c0::-moz-placeholder {
  color: #949ea3;
}

.c0:-ms-input-placeholder {
  color: #949ea3;
}

.c0::placeholder {
  color: #949ea3;
}

.c0:focus {
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
  border-color: #146292;
  box-shadow: 0 0 2px #146292 inset;
}

<Input__StyledInput
  onChange={[Function]}
  type="text"
  value="Some text"
>
  <input
    className="c0"
    onChange={[Function]}
    type="text"
    value="Some text"
  />
</Input__StyledInput>
`;

exports[`Input Component snapshots should match snapshot 3`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  padding-left: calc(0.5714285714285714rem - 1px);
  padding-right: calc(0.5714285714285714rem - 1px);
  padding-top: calc(0.625rem - 1px);
  padding-bottom: calc(0.625rem - 1px);
  border: solid 1px #e4e8ea;
  border-radius: 0.25rem;
  background-color: #e4e8ea;
  outline: none;
  color: #949ea3;
  font-size: 0.875rem;
  line-height: 1.1428571428571428;
  text-overflow: ellipsis;
  cursor: not-allowed;
}

.c0::-webkit-input-placeholder {
  color: #949ea3;
}

.c0::-moz-placeholder {
  color: #949ea3;
}

.c0:-ms-input-placeholder {
  color: #949ea3;
}

.c0::placeholder {
  color: #949ea3;
}

.c0:focus {
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
  border-color: #146292;
  box-shadow: 0 0 2px #146292 inset;
}

<Input__StyledInput
  disabled={true}
  placeholder="This input is disabled"
  type="text"
>
  <input
    className="c0"
    disabled={true}
    placeholder="This input is disabled"
    type="text"
  />
</Input__StyledInput>
`;

exports[`Input Component snapshots should match snapshot 4`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  padding-left: calc(0.5714285714285714rem - 1px);
  padding-right: calc(0.5714285714285714rem - 1px);
  padding-top: calc(0.625rem - 1px);
  padding-bottom: calc(0.625rem - 1px);
  border: solid 1px #d2d9dd;
  border-radius: 0.25rem;
  background-color: #ffffff;
  outline: none;
  color: #2d3b43;
  font-size: 0.875rem;
  line-height: 1.1428571428571428;
  text-overflow: ellipsis;
  cursor: text;
  border-color: #e05f6e;
}

.c0::-webkit-input-placeholder {
  color: #949ea3;
}

.c0::-moz-placeholder {
  color: #949ea3;
}

.c0:-ms-input-placeholder {
  color: #949ea3;
}

.c0::placeholder {
  color: #949ea3;
}

.c0:focus {
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
  border-color: #146292;
  box-shadow: 0 0 2px #146292 inset;
}

.c0:focus {
  border-color: #e05f6e;
  box-shadow: 0 0 2px #e05f6e inset;
}

<Input__StyledInput
  invalid={true}
  onChange={[Function]}
  placeholder="You could type something here"
  type="text"
  value="invalid value"
>
  <input
    className="c0"
    onChange={[Function]}
    placeholder="You could type something here"
    type="text"
    value="invalid value"
  />
</Input__StyledInput>
`;

exports[`Input Component snapshots should match snapshot 5`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  padding-left: calc(0.5714285714285714rem - 1px);
  padding-right: calc(0.5714285714285714rem - 1px);
  padding-top: calc(0.625rem - 1px);
  padding-bottom: calc(0.625rem - 1px);
  border: solid 1px #d2d9dd;
  border-radius: 0.25rem;
  background-color: #ffffff;
  outline: none;
  color: #2d3b43;
  font-size: 0.875rem;
  line-height: 1.1428571428571428;
  text-overflow: ellipsis;
  cursor: text;
}

.c0::-webkit-input-placeholder {
  color: #949ea3;
}

.c0::-moz-placeholder {
  color: #949ea3;
}

.c0:-ms-input-placeholder {
  color: #949ea3;
}

.c0::placeholder {
  color: #949ea3;
}

.c0:focus {
  -webkit-transition: 0.2s linear;
  transition: 0.2s linear;
  border-color: #146292;
  box-shadow: 0 0 2px #146292 inset;
}

<Input__StyledInput
  placeholder="You could type something here"
  type="text"
>
  <input
    className="c0"
    placeholder="You could type something here"
    type="text"
  />
</Input__StyledInput>
`;
